pkg/stub/handler.go

```diff
@@ -1117,6 +1127,13 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 					logrus.Debugf("Handle ignoring because config the same and exists is true and in progress false")
 					return nil
 				}
+				// if config changed, but a prior config action is still in progress,
+				// return error so that we enact this config change against a steady
+				// state system ... think changes to skipped imagestreams and which
+				// imagestreams to account for in the in progress tracking
+				if changed && srcfg.ConditionTrue(v1alpha1.ImageChangesInProgress) {
+					return fmt.Errorf("this config change will not be operated on until in progress from the prior config setting becomes false")
```

Comment:
rather than returning an error for retry, I'm considering setting in progress to false and clearing out the reason field ... then on the next event process the config change 

thoughts @bparees

Reply:
I definitely prefer an approach where we don't have to queue up changes but rather can just apply the latest desired config to whatever state we are in.  you'd have to tell me, though, if that can potentially cause issues where we're in the middle of creating samples(or something) and then we process a new config that, say, changes the registry for all the samples.. will we end up in the right final state?